File: benches/python/sum.py
#!/usr/bin/env python3
"""Sum benchmark for Python"""

def sum_n(n: int) -> int:
    result = 0
    i = 0
    while i <= n:
        result = result + i
        i = i + 1
    return result

def main():
    import sys
    n = int(sys.argv[1]) if len(sys.argv) > 1 else 10000
    result = sum_n(n)
    print(result)

if __name__ == "__main__":
    main()

File: benches/cpp/sum.cpp
// Sum benchmark for C++

#include <iostream>
#include <cstdlib>
#include <cstdint>

int64_t sum_n(int64_t n) {
    int64_t result = 0;
    int64_t i = 0;
    while (i <= n) {
        result = result + i;
        i = i + 1;
    }
    return result;
}

int main(int argc, char* argv[]) {
    int64_t n = argc > 1 ? std::atoll(argv[1]) : 10000;
    int64_t result = sum_n(n);
    std::cout << result << std::endl;
    return 0;
}

